compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -699,8 +706,10 @@ static void enforceLayoutToBroadcastOp(
 
   auto resultShape = broadcast.getResultVectorType().getShape();
   auto inputType = broadcast.getSourceType();
-  assert(isa<VectorType>(inputType) &&
-         "Scalar broadcast not supported for now.");
+  if (!isa<VectorType>(inputType)) {
+    return;
+  }
+
```

Comment:
Follow the llvm coding style and use `dyn_cast` here to avoid repeated type checking below